code/game/objects/items/supermatter_delaminator.dm

```diff
@@ -0,0 +1,213 @@
+	if(!istype(T) || !T.shard || shard || sealed)
+		return FALSE
+	T.shard.forceMove(src)
+	shard = T.shard
+	T.shard = null
+	T.update_icon()
+	update_icon()
+	to_chat(user, "<span class='warning'>Container is resealing...</span>")
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return TRUE
+
+/obj/item/antinoblium_container/proc/unload(obj/item/hemostat/antinoblium/T, mob/user)
+	if(!istype(T) || T.shard || !shard || sealed)
+		return FALSE
+	src.shard.forceMove(T)
+	T.shard = shard
+	shard = null
+	T.update_icon()
+	update_icon()
+	to_chat(user, "<span class='warning'>[user] gingerly takes out the antinoblium shard with the tongs...</span>")
+	return TRUE
+
+/obj/item/antinoblium_container/proc/seal()
+	if(sealed)
+		return
+	if(istype(shard) && !sealed)
+		STOP_PROCESSING(SSobj, shard)
+		playsound(src, 'sound/items/Deconstruct.ogg', 60, 1)
+		sealed = TRUE
+		update_icon()
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[src] is temporarily resealed, [shard] is safely contained.</span>")
+
+/obj/item/antinoblium_container/proc/unseal()
+	if(!sealed)
+		return
+	sealed = FALSE
+	update_icon()
+	if(ismob(loc))
+		to_chat(loc, "<span class='warning'>[src] is unsealed, the [shard] can now be removed.</span>")
+
+/obj/item/antinoblium_container/attackby(obj/item/hemostat/antinoblium/tongs, mob/user)
+	if(istype(tongs))
+		if(!tongs.shard)
+			unload(tongs, user)
+		else
+			load(tongs, user)
+	else
+		return ..()
+
+/obj/item/antinoblium_container/attack_self(mob/user)
+	if(sealed && shard)
+		unseal()
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[user] opens the [src] revealing the [shard] contained inside!</span>")
+	else if(!sealed && shard)
+		seal()
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[user] seals the [src].</span>")
+
+/obj/item/antinoblium_container/update_icon()
+	if(sealed)
+		icon_state = "antinoblium_container_sealed"
+	else if (shard)
+		icon_state = "antinoblium_container_loaded"
+	else
+		icon_state = "antinoblium_container_empty"
+	
+
+/obj/item/hemostat/antinoblium
+	name = "antinoblium extraction tongs"
+	desc = "A pair of tongs made from condensed hyper-noblium gas, searingly cold to the touch, that can safely grip an antinoblium shard."
+	icon = 'icons/obj/supermatter_delaminator.dmi'
+	icon_state = "antinoblium_tongs"
+	toolspeed = 0.75
+	damtype = "fire"
+	var/obj/item/supermatter_delaminator/antinoblium_shard/shard
+
+/obj/item/hemostat/antinoblium/Destroy()
+	QDEL_NULL(shard)
+	return ..()
+
+/obj/item/hemostat/antinoblium/update_icon()
+	if(shard)
+		icon_state = "antinoblium_tongs_loaded"
+	else
+		icon_state = "antinoblium_tongs"
+
+/obj/item/hemostat/antinoblium/afterattack(atom/O, mob/user, proximity)
+	. = ..()
+	if(!shard)
+		return
+	if(proximity && ismovable(O) && O != shard  && !istype(O, /obj/item/antinoblium_container) && !istype(O, /obj/machinery/power/supermatter_crystal))
+		Consume(O, user)
```

Comment:
note2self

Reply:
Resolved I guess? This is intended behaviour...